### ISSUE TITLE
MS SQL Server 2014: CSV import - timestamp values are shifted

### ISSUE THREAD

**Issue by keen85** [dbeaver/dbeaver#9777]
#### System information: 
- OS: Windows 10 Enterprise 
- DBeaver version: Version 7.2.0.202008302047
- Additional extensions: none

#### Connection specification:
- MS SQL Server 2014
- Driver name: mssql-jdbc-8.2.2.jre8.jar
- Do you use tunnels or proxies (SSH, SOCKS, etc)? No

#### Describe the problem you're observing:
I would like to import values from a CSV file. The CSV file contains timestamp values.
After the import, the **timestamps from the CSV file are shifted** in the database - some by +2 hours, some by +1 hour some even shift the date +2 days.
I am located in my timezone European Central - if this might have any influence.

#### Steps to reproduce, if exist:
DDL for creating a table

> DROP TABLE IF EXISTS dbo.DBEAVER_IMPORT_TEST;
> CREATE TABLE dbo.DBEAVER_IMPORT_TEST (
>   INT_ATTRIBUTE    SMALLINT,
>   STR_ATTRIBUTE    varchar(100),
>   TS_ATTRIBUTE     datetime2(0)
> )
> ;

CSV file

> INT_ATTRIBUTE;STR_ATTRIBUTE;TS_ATTRIBUTE
> 1;"Hello World";***2020-09-15 14:49:00***
> 2;Hello World;***1970-01-01 02:00:00***
> 3;"Hello ;World";***0001-01-01 00:00:00***
> 4;"Hello \;World";***0001-01-01 00:00:00***
> 5;          ;***0001-01-01 00:00:01***
> 6;"Hello \"World";***0001-01-01 00:00:01***

Import settings:
![image](https://user-images.githubusercontent.com/29750255/93216124-a202b500-f767-11ea-898d-ba11f6d7ba44.png)

Preview already shows the timeoffset:
![image](https://user-images.githubusercontent.com/29750255/93216658-484eba80-f768-11ea-999b-e6db45376200.png)

Result in DB
![image](https://user-images.githubusercontent.com/29750255/93216340-e7bf7d80-f767-11ea-8f77-bca0569ffc83.png)

**Comment by kseniiaguzeeva** [comment]
Thank you for the report. Could you please send me also Data Formats settings (Window>Preferences>Database Editors>Data Editor>Data Formats)?
![dbeaver_4et05G32dc](https://user-images.githubusercontent.com/51405061/93571713-8385ff00-f99d-11ea-8ba5-9ce38a396737.png)

**Comment by keen85** [comment]
Hello @kseniiaguzeeva,
thanks for your response.

This are the connection settings
![image](https://user-images.githubusercontent.com/29750255/93575257-b5956200-f999-11ea-8d70-aafebfb5773b.png)

and the global ones:
![image](https://user-images.githubusercontent.com/29750255/93575217-aa423680-f999-11ea-9489-133847fc284c.png)

**Comment by serge-rider** [comment]
1. Problem with historical date shifts

This happens with all databases when importing from CSV/XLSX.
It happens because date like 0001-01-01 are historical, i.e. had place before Gregorian calendar was instantiated.
See details at https://stackoverflow.com/a/23978037/1912589
001-01-01 was converted into 0001-01-03 during LocalDateTime->Date transformation.
I've fixed it, now we will use java.sql.Timestamp transform function.

2. Problem with hours shifting.

This usually happens because your local time zone differs from the server time zone. By default DBeaver inserts data in the local timezone which then converted to the server one.

Anyway, to cover such cases I've added new property to CSV/XLSX importers:
![image](https://user-images.githubusercontent.com/6398845/98466259-6cb29a80-21df-11eb-859b-fdcd62c600b8.png)
